src/sentry/uptime/models.py

```diff
@@ -264,5 +264,17 @@ def get_top_hosting_provider_names(limit: int) -> set[str]:
     )
 
 
+@cache_func_for_models(
+    [(UptimeSubscriptionRegion, lambda region: (region.uptime_subscription_id,))],
+    recalculate=False,
+)
+def get_regions_for_uptime_subscription(
+    uptime_subscription_id: int,
+) -> list[UptimeSubscriptionRegion]:
+    return list(
+        UptimeSubscriptionRegion.objects.filter(uptime_subscription_id=uptime_subscription_id)
+    )
```

Comment:
I'm going to add caches like this for other things we fetch, hoping it might improve performance here a little.